### PR TITLE
Update to React 15.5

### PR DESCRIPTION
**Pull request by jbalboni**
Connects to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/2042
Connects to https://github.com/department-of-veterans-affairs/vets-website/issues/3542

This updates us to React 15.5. It also makes some minor updates to other packages, and major version updates to react-router and skin-deep (major in the semantic versioning sense, not the impact sense).

This change creates some warnings in the console and in the unit test output. Some of those will go away when we update our prop types. I'm going to leave that to a separate issue, because it touches every component.

### CODE REVIEW

File: test/common/components/SampleFeature.unit.spec.jsx
Comment (jbalboni) [inline test/common/components/SampleFeature.unit.spec.jsx]: skin-deep 1.0 seems more strict about shallow rendering, and wasn't rendering `<SampleFeature/>` as it was written before.

File: test/edu-benefits/1990/components/SubmitMessage.unit.spec.jsx
Comment (cvalarida) [inline test/edu-benefits/1990/components/SubmitMessage.unit.spec.jsx]: This seems like an odd change for this PR. What prompted it? Something with skin-deep 1.0, or should this have been `<SubmitMessage>` already?

File: test/common/components/SampleFeature.unit.spec.jsx
Comment (cvalarida) [inline test/common/components/SampleFeature.unit.spec.jsx]: This is perhaps a result of their introducing the [dive](https://www.npmjs.com/package/skin-deep#going-deeper) command?

File: package.json
Comment (cvalarida) [inline package.json]: Looks like this is [being depreciated](https://facebook.github.io/react/blog/2017/04/07/react-v15.5.0.html#react-test-utils) in favor of `react-dom/test-utils`. We'll have to change...I think all our unit tests to import that instead.

**Review by cvalarida — APPROVED** [review]
Looks good. Just had a question (mostly for my curiosity). The depreciation of `react-addons-test-utils` isn't a deal-breaker, but I think we should migrate while we're already upgrading.

File: test/edu-benefits/1990/components/SubmitMessage.unit.spec.jsx
Comment (jbalboni) [inline test/edu-benefits/1990/components/SubmitMessage.unit.spec.jsx]: It should have been that already.

File: test/common/components/SampleFeature.unit.spec.jsx
Comment (jbalboni) [inline test/common/components/SampleFeature.unit.spec.jsx]: That was there before, I think.

File: package.json
Comment (jbalboni) [inline package.json]: Yep, I meant to look for a codemod to update that, but forgot.

Comment (jbalboni) [inline package.json]: I think I'm going to leave this to another PR (along with the prop types change). It'll touch so many components and I wanted to keep this PR as simple as possible.

**Review by bshyong — APPROVED** [review]
LGTM let's get this one merged before I merge in the webpack 2 changes

**Pull request merged**
